File: LeetCode - Algorithm/Stack and Queue/71.py
# 71. Simplify Path
# https://leetcode.com/problems/simplify-path/


class Solution:
    def simplifyPath(self, path: str) -> str:
        resultPathes = []

        originalPathes = path.split("/")

        for pathElem in originalPathes:
            if pathElem == "":
                continue
            elif pathElem == ".":
                continue
            elif pathElem == "..":
                if resultPathes:
                    resultPathes.pop()
            else:
                resultPathes.append(pathElem)

        result = "/"

        for index, pathElem in enumerate(resultPathes):
            result += pathElem

            if index < len(resultPathes) - 1:
                result += "/"

        return result


"""
The hardest part of this problem is recognizing that it's a 'Stack' problem.
"""

File: LeetCode - Algorithm/Stack and Queue/71.js
// 71. Simplify Path
// https://leetcode.com/problems/simplify-path/

/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function (path) {
  const resultStack = [];

  const pathDivided = path.split("/");

  for (let i = 0; i < pathDivided.length; i++) {
    const current = pathDivided[i];
    if (current === "") continue;
    else if (current === ".") continue;
    else if (current === "..") resultStack.pop();
    else resultStack.push(current);
  }

  return "/" + resultStack.join("/");
};

/*
path = "/home/user/Documents/../Pictures"
*/
